.github/workflows/run-tests.yml

```diff
@@ -264,6 +265,14 @@ jobs:
           TEST_ARGS: ${{ needs.set-up-single-test.outputs.single_test_args }}
           TEST_TIMEOUT: ${{ needs.set-up-single-test.outputs.test_timeout }}
 
+      - name: Generate crash report
+        if: failure()
+        run: |
+          xml_count=$(compgen -G ".testoutput/junit.*.xml" | wc -l || echo 0)
+          if [ "$xml_count" -lt "${TEST_ATTEMPTS:-2}" ]; then
+            CRASH_REPORT_NAME="${GITHUB_JOB}" make report-test-crash
+          fi
```

Comment:
New step that checks if there was a crash; and then emits the JUnit crash report file.